parsing/parser.mly

```diff
@@ -2433,7 +2440,7 @@ let_binding_body:
         let typ = ghtyp ~loc (Ptyp_poly([],t)) in
         let patloc = ($startpos($1), $endpos($2)) in
         (ghpat ~loc:patloc (Ppat_constraint(v, typ)),
-         mkexp_constraint ~loc:$sloc $4 $2) }
+         mkexp_constraint ~loc:$sloc $4 $2, false) }
```

Comment:
We avoid these `false` blemishes in our semantic actions with the following setup:

```
let_binding_body:
| let_binding_body_with_punning { let (p,e) = $1 in (p, e, true) }
| let_binding_body_no_pun { let (p, e) = $1 in (p, e, false) }
```

Reply:
That is better, I'll change to something along those lines. (I don't think `let_binding_body_with_punning` is necessary as a separate nonterminal, though, as there's only one case)